server/controllers/finance/reports/balance/index.js

```diff
@@ -172,76 +173,52 @@ function balanceReporting(params) {
   const query = params;
   const data = {};
 
-  const hasClasse = (query.classe !== '*');
-  const dateRange = (query.dateFrom && query.dateTo);
-
-  // gets the amount up to the current period
-  let sql = `
-    SELECT a.number, a.id, a.label, a.type_id, a.is_charge, a.is_asset, SUM(pt.credit) AS credit, SUM(pt.debit) AS debit
-    FROM period_total AS pt JOIN account AS a ON pt.account_id = a.id
-    JOIN period AS p ON pt.period_id = p.id
-    WHERE p.end_date <= DATE(?) AND pt.enterprise_id = ?
-    ${hasClasse ? 'AND a.classe = ? ' : ''}
-    GROUP BY a.id `;
-
-  if (dateRange) {
-    sql = `
-    SELECT a.number, a.id, a.label, a.type_id, a.is_charge, a.is_asset, SUM(pt.credit) AS credit, SUM(pt.debit) AS debit
-    FROM period_total AS pt JOIN account AS a ON pt.account_id = a.id
-    JOIN period AS p ON pt.period_id = p.id
-    WHERE p.start_date >= DATE(?) AND start_date < DATE(?) AND pt.enterprise_id = ?
-     ${hasClasse ? 'AND a.classe = ? ' : ''}
-    GROUP BY a.id `;
-  }
+  const sqlFiscalYear = `
+    SELECT id, number_of_months FROM fiscal_year WHERE DATE(?) BETWEEN DATE(start_date) AND DATE(end_date) LIMIT 1;
+  `;
+
+  let sql;
+  let fiscal;
 
-  const queryParameters = (dateRange) ?
-    [query.dateFrom, query.dateTo, query.enterpriseId, query.classe] :
-    [query.date, query.enterpriseId, query.classe];
-
-  return db.exec(sql, queryParameters)
-  .then((rows) => {
-    data.beginning = rows;
-
-    sql = `
-      SELECT a.number, a.label, a.id, a.type_id, a.is_charge, a.is_asset,
-        SUM(pt.credit) AS credit, SUM(pt.debit) AS debit
-      FROM period_total AS pt JOIN account AS a ON pt.account_id = a.id
-      JOIN period AS p ON pt.period_id = p.id
-      WHERE DATE(?) BETWEEN p.start_date AND p.end_date AND pt.enterprise_id = ?
-       ${hasClasse ? 'AND a.classe = ? ' : ''}
-      GROUP BY a.id;`;
-
-    query.date = (dateRange) ? query.dateTo : query.date;
-
-    return db.exec(sql, [query.date, query.enterpriseId, query.classe]);
-  })
-  .then((rows) => {
-    data.middle = rows;
-    return data;
-  })
-  .then(() => {
-    // Manual mixing
-
-    // fill with zero if all accounts
-    sql =
-      `SELECT a.number, a.id, a.label, a.type_id, a.is_charge, a.is_asset, '0' AS credit, '0' AS debit
-       FROM account a WHERE a.type_id <> ${TITLE_ACCOUNT_TYPE} AND a.locked = 0;`;
-
-    return query.accountOption === 'all' ? db.exec(sql) : false;
-  })
-  .then((rows) => {
-    if (!rows) { return data; }
-
-    // Naive manipulation for filling with zero
-    const accounts = rows;
-    const touched = data.beginning.map(item => item.id);
-
-    accounts.forEach(item => {
-      if (touched.indexOf(item.id) === -1) {
-        data.beginning.push(item);
+  return db.one(sqlFiscalYear, [query.date])
+    .then((fiscalYear) => {
+      if (!fiscalYear.id) {
+        throw new NotFound(`Could not find a Fiscal Year for the date ${query.date}.`);
```

Comment:
I believe that `db.one()` makes this error unnecessary.  If no record is found, it will throw and error.